backends/tc/midend.cpp

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright (C) 2023 Intel Corporation
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions
+and limitations under the License.
+*/
+
+#include "midend.h"
+
+namespace TC {
+
+/**
+This class implements a policy suitable for the ConvertEnums pass.
+The policy is: convert all enums that are not part of the v1model.
+Use 32-bit values for all enums.
+*/
+class EnumOn32Bits : public P4::ChooseEnumRepresentation {
+    bool convert(const IR::Type_Enum *type) const override {
+        if (type->srcInfo.isValid()) {
+            return true;
+        }
+        return true;
+    }
+    unsigned enumSize(unsigned) const override { return 32; }
+};
+
+const IR::ToplevelBlock *MidEnd::run(TCOptions &options, const IR::P4Program *program,
```

Comment:
Does it differ from the midend in the eBPF backend? We could re-use that probably.

Reply:
I think it is better to keep separate Midend for tc since we might need to add different passes in future. Reusing midend from ebpf backend might not be ideal when passes need to be added in future

Reply:
ok